airflow/executors/knative_executor.py

```diff
@@ -0,0 +1,186 @@
+            log.info(resp.status)
+            log.info(await resp.text())
+            return resp
+
+
+class KnativeRequestLoop(multiprocessing.Process, LoggingMixin):
+    def __init__(self,
+                 task_pipe,
+                 result_pipe,
+                 pop_running_pipe,
+                 host_header,
+                 host,
+                 ):
+
+        super().__init__()
+        self.host = host
+        self.task_pipe = task_pipe
+        self.result_pipe = result_pipe
+        self.host_header = host_header
+        self.pop_running_pipe = pop_running_pipe
+
+    def recieve_and_execute(self, loop):
+        key = self.task_pipe.recv()
+        loop.create_task(self.execute_work(key=key))
+
+    def run(self):
+        loop = asyncio.get_event_loop()
+        loop.add_reader(fd=self.task_pipe, callback=functools.partial(self.recieve_and_execute, loop))
+        loop.run_forever()
+
+    async def execute_work(self, key):
+        """
+        Executes command received and stores result state in queue.
+        :param task_instance:
+        :param key: the key to identify the TI
+        :type key: tuple(dag_id, task_id, execution_date)
+        :param command: the command to execute
+        :type command: str
+        """
+        if key is None:
+            return
+        self.log.info("%s running %s", self.__class__.__name__, key)
+        (dag_id, task_id, execution_date, try_number) = key
+        try:
+            future = make_request_async(
+                task_id,
+                dag_id,
+                execution_date,
+                host=self.host, log=self.log, host_header=self.host_header)
+            resp = await future
+            if resp.status != 200:
+                state = State.FAILED
+                self.log.error("Failed to execute task %s.", str(resp.text))
+                self.result_pipe.send((key, state))
+            else:
+                self.log.info("assuming task success")
+                # self.result_queue.put((key, None))
+                self.pop_running_pipe.send(key)
+
+        except asyncio.InvalidStateError as e:
+            state = State.FAILED
+            self.log.error("Failed to execute task %s.", str(e))
+            self.result_pipe.send((key, state))
+
+
+class KnativeExecutor(BaseExecutor):
+    """
+    KnativeExecutor executes tasks locally in parallel. It uses the
+    multiprocessing Python library and queues to parallelize the execution
+    of tasks.
+    """
+    task_queue = None
+    result_queue = None
+    result_pipe = None
+    pop_running_queue = None
+
+    def terminate(self):
+        pass
+
+    @cached_property
+    def kube_executor(self):
+        self.kube_executor_initialized = True
+        return KubernetesExecutor()
+
+    def __init__(self):
+        self.kube_executor_initialized = False
+        super().__init__()
+
+    def start(self):
+        req = conf.get("knative", "knative_host")
```

Comment:
Can the config option just be called `host` please (for example we renamed elasticsearch from `elasticsearch_host` to just `host` as the section qualifies it.)